src/app/content/utils.ts

```diff
@@ -1,8 +1,37 @@
 import { HTMLElement } from '@openstax/types/lib.dom';
+import flatten from 'lodash/fp/flatten';
+import { isArchiveTree } from './guards';
+import { ArchiveTree, ArchiveTreeSection } from './types';
 
 export const stripIdVersion = (id: string): string => id.split('@')[0];
 export const getIdVersion = (id: string): string | undefined => id.split('@')[1];
 
+export const getContentPageReferences = (content: string) =>
+  (content.match(/\/contents\/([a-z0-9\-]+(@[\d\.]+)?)(:([a-z0-9\-]+(@[\d\.]+)?))?/g) || [])
```

Comment:
I am worried about extracting links by executing a regular expressions over the whole HTML page. Why are these resolved before injecting the HTML into the DOM rather than after?

Reply:
its being applied to the content string before being loaded into the dom (differentiating from `HTML page` which could mean the whole page, not just the content). its happening this way so that we can load content references as part of the initial data load before anything is rendered. this has some benefits:

- modified links are pre-rendered
- there is no interstitial time where a user could click a link before it gets replaced

Reply:
Okies, I just hope no one writes something like:

`<p>To find out more, you can visit <a href="/contents/{uuid}">/contents/foobar</a> or just type /contents/foobar into your browser`

Reply:
the regex isn't looking for a link, its looking for the paths, so that would actually work 😄

Reply:
that was my original concern 😄.

RE interstitial time: I see what you mean... it _could_ be alleviated by loading the HTML into a detached DOM node, finding the links, rewriting them, and then attaching it to the DOM but maybe that's overkill?